### DISCUSSION TITLE
How to modify number of inventory "pop ups screen" when comparing between items

### DISCUSSION THREAD

**Discussion by Danimal696** [flareteam/flare-game#892] [Q&A]
I use 4 rings intead of 2, when comparing with unequiped item inside inventory, only the first 2 show their stats. Can the remainings ones be shown as well?

**Comment by WithinAmnesia** [comment]
If there are 10 rings equipped what does the engine do then XD? A player character could have 3+ of the same item type equipped. Is this a hard / fixed value of how many current equipped items of the same item type can be 'popped up' to show a comparison? I wonder what the complexity of that is or if there is a simpler way to compare lots of items equipped of the same type without making lots of complexity to implement? A cycle compare button or like hold shift to compare to avoid massive pop up spam (if there ~10 pop up item compare windows when mousing over a ~ring) when organizing an inventory? Maybe highlight the equipped item slots being compared if more than 2 cannot be enlarged to hopefully clue in the player they can cycle items in those slots to still effectively compare more than 2 equipped items of the same type but only 2 at a time from a designated area? There is probably a simpler solution to solve this that is not too complex yet I am not sure what scale of complexity is behind this; if it is simple or if it is more complex than easily observed. I think the item compare is really cool either way and trying to expand it is a neat idea.

**Comment by dorkster** [comment]
We had a fixed value of 3 for the maximum number of visible floating tooltips (1 normal + 2 comparison). Since it's pretty trivial, I've [updated the engine](https://github.com/flareteam/flare-engine/commit/24d48314139241da63fc6fc32249aed2dcc6047a) to expose this value as part of `engine/tooltips.txt`. So if you want to compare an item to all 4 rings, you'll be able to use `tooltip_visible_max=5`.

Keep in mind that these tooltips are only arranged in a horizontal line, so they might not all fit on screen. Hence why I had set the limit to 3.

**Reply by WithinAmnesia** [reply]
Okay so I take it that max=11 would be a bit much XD 'okay so what does this ring do' FOOMP! *a line of pop ups reaching across the screen and possibly beyond* XD. I guess it could be tinkered with to see what is possible within reason. I guess the floating tooltip boxes could be changed in size to fit more on screen maybe as ~'something' related to how many tool tips can reasonably fit on screen XD. I guess also this is up to a balance of finding what is informative verses what is unneeded screen clutter. It is cool that this whole floating tool tip maximum number can be tweaked with by the developers arbitrary discretion game by game (within reason).

**Reply by Danimal696** [reply]
Thanks Dorkster, where can I grab a build with the change applied somewhere?, though don´t do it just for me, I can wait for a future build with more new features.

Whitinamnesia, you need to tone down some of your design decisions, you know, just adding two more rings makes balancing enemies way harder. Now I need to take into account not just the base power (att +def) the player has at a level but add 4 rings of variation as well, that means to compensate for overpowered items the players might find, enemies need to be stronger, meaning unlucky players will have it way harder, specially the magician class which are crystal cannons. With 10 rings? you will need to make the rings so worthless they will barely affect the players to compensate for the numbers you can wear.

Normally less is more, so i advice you to start small and build from there.

**Reply by WithinAmnesia** [reply]
@Danimal696 Well that depends on how a person sets up the source of the player power granted by ~'magically enchanted' / 'magical' / 'magic' items. If there is no cost to the player for a bonus then it is like free power and yes, that can run away on game balance and become a very slippery slope. In my case I am trying to follow a simple principle with my game design, what would real life ~'allow' within reason for these ~'magical' / 'magic' items to operate in the world; almost like a science where nothing is exactly ~'free'? So with these magic rings sure they can harbour 'magic' / 'enchantments' yet at what cost? A tank in real life needs fuel, ammunition, maintenance, careful strategic operation to successful use even for just a day. Now I don't want to be too crazy and ruin the potential fun of a medieval fantasy world with the banalities of real life. We all endure them perhaps more so than we can already handle let alone bogging down a hopefully enjoyable game experience.

So back to the ~'10 rings'. Sure maybe you could put 10 rings on your person; but at what cost? If the magic benefit is 'free' then yeah it is ungrounded high fantasy and things can get out of hand real fast for game balance. Yet what happens when those magical rings demand something in return to operate and force the player to balance cost and benefit just like in real life? What if it does not matter how big an item is (within reason) to harbour significant magic energies? What happens when a player cannot 'afford' the ~'power' demand of high power ~'magical' items? Why are historical swords not 200 pounds and 100 feet long? Well simple the operator could not afford to operate the muscle power in order to operate such a combat system from their own body power within reason. What if ~'magic' operated like that? Ever see a singer perform so hard that it is small wonder they are not half dead at the end of a concert? Sure they are just singing at first glance but deep down they need to summon every bit of ~'spirit' / 'character' out of themselves to really drive home to deliver / realize their passions / visions of what they desire. It is like they are battling themselves and the more a singer can 'harness' from their 'heart' / 'soul' / 'spirit' the farther they can explore their musical ability (within reason). The same is true with boxing and or any high demanding activity where you lay your heart and soul out and push yourself to your absolute limit. Then even when the odds of life and adversity are against yourself your 'spirit' / 'soul' can ~'animate' you beyond your wildest of expectations; if given enough opportunity. So I ask why not take this approach with 'magical' / 'magic' items? Flashlights have batteries, they don't last forever if the charge is not replenished. Why not have a 'finite' forms of energies in a medieval fantasy setting?

If I can get available ~'-% to resource regen' then maybe having 10 high power 'magical' rings equipped would by default result in ~'-150% Mana / Spirit / Health Regeneration' (effectively -50% _degeneration_)[without something to offset the regeneration deficiency / cost inflicted upon the character]? Then maybe one could wear the 10 high power rings but at what cost? How is the player going to deal with that? What options are available? Could they even ~'afford' to wear any other 'magical' / 'magic' items? What if ~'power' negotiated by a character is a finite resource, just like in real life; yet hopefully more diverse and intriguing within a medieval fantasy setting? Perhaps it could be true that less is more? When a person has to pay a price for what they own; then all of a sudden things become valuable in as so far as the cost inflicted upon said person in order for them to pay for something to own.

**Reply by dorkster** [reply]
@Danimal696 If you don't mind waiting, 1.13 will be released on Jan 31. Until then, here's a screenshot of 5 visible tooltips:

![Screenshot_20220125_094046](https://user-images.githubusercontent.com/913982/150998255-7a1c6037-7dbb-47bd-a18e-b8af2b35bbcd.png)

As you can see, it starts to get a little unruly at this point. I think the solution to this problem might be to only show 1-2 comparison tooltips at a time and cycle through them on a timer.

**Reply by Danimal696** [reply]
Sure no problem waiting : ) 
Yeah, it´s a bit ugly but it should do, no need to complicate things further.

**Reply by WithinAmnesia** [reply]
A timer sounds like a simple and robust solution if there are more than 5-6 item comparison floating tool tips. I wonder if a timer option could be tinkered with like time delay before cycling, how many max pop ups / floating tool tips etc. could be tweaked with by game developers? Yet complexity to implement could be getting a bit much. So I feel an 'open end' / being flexible to find a solution could be more understanding in terms of dealing with potential complexity; bit by bit :-).
